superdesk/core/mongo.py

```diff
@@ -240,104 +247,138 @@ def stop(self):
         """Get a collection connection from a registered resource
 
         Caches the database connection based on the ``resource_name``, so subsequent calls re-use the same
         connection.
 
+        :param resource_name: The name of the registered resource
+        :param versioning: If ``True``, will provide client to the versioned collection
         :raises KeyError: if a resource with the provided ``resource_name`` is not registered
         """
 
-        return self.get_db(resource_name).get_collection(resource_name)
+        return self.get_db(resource_name, versioning).get_collection(
+            self.get_collection_name(resource_name, versioning)
+        )
 
     def create_resource_indexes(self, resource_name: str, ignore_duplicate_keys=False):
         """Creates indexes for a resource
 
-        If the resource config has ``versioning == True``, then a second index with suffix ``_versions``
-        will be created.
+        If the resource config has ``versioning == True``, then indexes will also be created for the
+        versioning collection.
 
+        :param resource_name: The name of the registered resource
+        :param ignore_duplicate_keys: If ``True``, will ignore duplicate key errors
         :raises KeyError: if a resource with the provided ``resource_name`` is not registered
         """
 
-        resource_config = self.get_resource_config(resource_name)
-        db = self.get_client(resource_name)[1]
-        collection_names = (
-            [resource_name] if not resource_config.versioning else [resource_name, f"{resource_name}_versions"]
-        )
-
-        for collection_name in collection_names:
-            collection = db.get_collection(collection_name)
-            if resource_config.indexes is None:
-                continue
-            for index_details in resource_config.indexes:
-                keys = [
-                    # (key.name, key.direction) if key.direction is not None else key.name
-                    (key[0], key[1])
-                    for key in index_details.keys
-                ]
-                kwargs = {key: val for key, val in asdict(index_details).items() if key != "keys" and val is not None}
-
-                try:
+        mongo_config = self.get_resource_config(resource_name)
+        indexes = mongo_config.indexes or []
+        if indexes:
+            self.create_collection_indexes(
+                self.get_collection(resource_name, versioning=False),
+                indexes,
+                ignore_duplicate_keys,
+            )
+
+        if mongo_config.versioning:
+            indexes = (mongo_config.version_indexes or []) + [
+                MongoIndexOptions(
+                    name="_id_document_1",
+                    keys=[(VERSION_ID_FIELD, 1)],
+                    background=True,
+                    unique=False,
+                ),
+                MongoIndexOptions(
```

Comment:
imo this one might be enough, that should be usable also for lookups using just `_id_document`